src/agent/src/main.rs

```diff
@@ -418,8 +418,15 @@ async fn start_sandbox(
 }
 
 // Start-up attestation-agent, CDH and api-server-rest if they are packaged in the rootfs
-fn init_attestation_components(logger: &Logger, _config: &AgentConfig) -> Result<()> {
-    // The Attestation Agent will run for the duration of the guest.
+// and the corresponding procs are enabled in the agent configuration. the process will be
+// launched in the background and the function will return immediately.
+fn init_attestation_components(logger: &Logger, config: &AgentConfig) -> Result<()> {
```

Comment:
I think we also need to review L406-408 (sorry - github still won't let us comment on unchanged lines) as that checks both CDH & AA paths exist before launching this code, whereas I guess now a user sets `agent.guest_components_procs=attestation-agent` then it would be valid to have a rootfs without CDH binary it and still launch it? I'm wondering if the best approach is to pull the path check(s) into this function?

Reply:
good point, forgot those. curiously only the CDH and AA bins presence was checked, ASR isn't.

It's tricky, because I assume the intention behind this clause is to make sure attestation init is only called on CoCo guest vms. There doesn't seem to be a coco-specific cargo build feature, the build-time toggle is probably a make target like `rootfs-initrd-confidential-tarball` that will add a coco-layer on the rootfs, but the agent is generic.

So, to keep the behaviour as-is for non-coco guests, we probably need to keep a check like this, as we default to `_procs=api-server-rest`. It's not great because this check is redundant, but I wouldn't know a better way.

Reply:
Yeah - I agree - In order to not have to have a separate agent build we effectively use that as a switch for confidential vs non-confidential VMs. Maybe we could just check the attestation-agent binary before calling init-attestation as that would allow the `agent.guest_components_procs=attestation-agent` to still work if only the attestation-agent is in the rootfs?

Reply:
I added a fn that checks for the presence of all the bins as a pre-requisite. is there a scenario in which we expect the AA bin to be present, but not CDH and ASR?

Reply:
Good question - with our rootfs builds no, but as we provide the option `agent.guest_components_procs=attestation-agent` to only launch the attestation I'm guessing there is a scenario that makes sense to only use the attestation-agent (though this might change if we flip the image-rs guest pull to happen through CDH instead) and based on that a downstream rootfs builder might optimise to only build in the AA? However you know the attestation systems better than I do, so I won't block this PR based on this issue.

Reply:
I see, I extended the verify logic to be a bit more forgiving and check only the bins that have been requested for launch

Reply:
Ok, that is obviously (but not to me!) the best way to do this. Thanks!